### PR TITLE
Add C API for WaitForCompact

### PR DESCRIPTION
**Pull request by abulimov**
Add a bunch of C API functions to expose new `WaitForCompact` function and related options.

Test Plan: unit tests

### CODE REVIEW

**Comment by akankshamahajan15** [comment]
@abulimov  Thank you for this change. Asan job is failing with memory leaks. Can you take a look? Thanks

File: db/c_test.c [hunk 1]
```diff
@@ -3687,6 +3687,47 @@ int main(int argc, char** argv) {
...
+    CheckCondition(1 == rocksdb_wait_for_compact_options_get_close_db(wco));
+
+    rocksdb_wait_for_compact_options_set_timeout(wco, 342);
+    CheckCondition(342 == rocksdb_wait_for_compact_options_get_timeout(wco));
+
+    rocksdb_wait_for_compact_options_destroy(wco);
+  }
+  StartPhase("wait_for_compact");
+  {
+    rocksdb_close(db);
```
Comment (jaykorean) [inline db/c_test.c:3712]: While the memory leak detected by ASAN test doesn't appear to be related to your change, I think you can still avoid unnecessary destroying / re-opening db here to avoid that error.

You can also add `rocksdb_wait_for_compact()` call in `CheckCompaction()` and see if that impacts any other tests.

Comment (abulimov) [reply db/c_test.c]: Thanks for the feedback, added `rocksdb_wait_for_compact()` to `CheckCompaction()` and removed reopening of db.

**Review by jaykorean — APPROVED** [review]
LGTM. Thanks!
